examples/dataflow-streaming-benchmark/src/main/java/com/google/cloud/pso/pipeline/StreamingBenchmark.java

```diff
@@ -0,0 +1,205 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pso.pipeline;
+
+import com.github.vincentrussell.json.datagenerator.JsonDataGeneratorException;
+import com.github.vincentrussell.json.datagenerator.impl.JsonDataGeneratorImpl;
+import com.google.common.collect.Maps;
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.Map;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.fs.MatchResult.Metadata;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.joda.time.Duration;
+
+/**
+ * The {@link StreamingBenchmark} is a streaming pipeline which generates messages at a specified
+ * rate to a Pub/Sub topic. The messages are generated according to a schema template which
+ * instructs the pipeline how to populate the messages with fake data compliant to constraints.
+ *
+ * <p>The number of workers executing the pipeline must be large enough to support the supplied QPS.
+ * Use a general rule of 2,500 QPS per core in the worker pool.
+ *
+ * <p>See <a href="https://github.com/vincentrussell/json-data-generator">json-data-generator</a>
+ * for instructions on how to construct the schema file.
+ *
+ * <p><b>Pipeline Requirements</b>
+ *
+ * <ul>
+ *   <li>The schema file exists.
+ *   <li>The Pub/Sub topic exists.
+ * </ul>
+ *
+ * <p><b>Example Usage</b>
+ *
+ * <pre>
+ * # Set the pipeline vars
+ * PROJECT_ID=PROJECT_ID
+ * PIPELINE_FOLDER=gs://${PROJECT_ID}/dataflow/pipelines/streaming-benchmark
+ *
+ * # Set the runner
+ * RUNNER=DataflowRunner
+ *
+ * # Build the template
+ * mvn compile exec:java \
+ * -Dexec.mainClass=com.google.cloud.pso.pipeline.StreamingBenchmark \
+ * -Dexec.cleanupDaemonThreads=false \
+ * -Dexec.args=" \
+ * --project=${PROJECT_ID} \
+ * --stagingLocation=${PIPELINE_FOLDER}/staging \
+ * --tempLocation=${PIPELINE_FOLDER}/temp \
+ * --runner=${RUNNER} \
+ * --zone=us-east1-d \
+ * --autoscalingAlgorithm=THROUGHPUT_BASED \
+ * --maxNumWorkers=50 \
+ * --qps=100000 \
+ * --schemaLocation=SCHEMA_LOCATION \
+ * --topic=TOPIC"
+ * </pre>
+ */
+public class StreamingBenchmark {
+
+  /**
+   * The {@link Options} class provides the custom execution options passed by the executor at the
+   * command-line.
+   */
+  public interface Options extends PipelineOptions {
```

Comment:
Can this be private?

Reply:
I normally leave Options public so that someone can programmatic-ly invoke the class by constructing the Options object and passing them to the run() method.